ProcessLib/RichardsComponentTransport/RichardsComponentTransportFEM-impl.h

```diff
@@ -0,0 +1,352 @@
+            sm.N.transpose() * sm.N * integration_factor *
+                _integration_method.getWeightedPoint(ip).getWeight());
+    }
+}
+
+template <typename ShapeFunction, typename IntegrationMethod,
+          unsigned GlobalDim>
+void LocalAssemblerData<ShapeFunction, IntegrationMethod, GlobalDim>::assemble(
+    double const t,
+    std::vector<double> const& local_x,
+    std::vector<double>& local_M_data,
+    std::vector<double>& local_K_data,
+    std::vector<double>& local_b_data)
+{
+    auto const local_matrix_size = local_x.size();
+    // This assertion is valid only if all nodal d.o.f. use the same shape
+    // matrices.
+    assert(local_matrix_size == ShapeFunction::NPOINTS * NUM_NODAL_DOF);
+
+    auto local_M = MathLib::createZeroedMatrix<LocalMatrixType>(
+        local_M_data, local_matrix_size, local_matrix_size);
+    auto local_K = MathLib::createZeroedMatrix<LocalMatrixType>(
+        local_K_data, local_matrix_size, local_matrix_size);
+    auto local_b = MathLib::createZeroedVector<LocalVectorType>(
+        local_b_data, local_matrix_size);
+
+    unsigned const n_integration_points =
+        _integration_method.getNumberOfPoints();
+
+    SpatialPosition pos;
+    pos.setElementID(_element_id);
+
+    auto p_nodal_values = Eigen::Map<const NodalVectorType>(
+        &local_x[pressure_index], pressure_size);
+
+    auto const& b = _process_data.specific_body_force;
+
+    MaterialLib::Fluid::FluidProperty::ArrayType vars;
+
+    GlobalDimMatrixType const& I(
+        GlobalDimMatrixType::Identity(GlobalDim, GlobalDim));
+
+    auto KCC = local_K.template block<concentration_size, concentration_size>(
+        concentration_index, concentration_index);
+    auto MCC = local_M.template block<concentration_size, concentration_size>(
+        concentration_index, concentration_index);
+    auto Kpp = local_K.template block<pressure_size, pressure_size>(
+        pressure_index, pressure_index);
+    auto Mpp = local_M.template block<pressure_size, pressure_size>(
+        pressure_index, pressure_index);
+    auto Bp = local_b.template block<pressure_size, 1>(pressure_index, 0);
+
+    for (std::size_t ip(0); ip < n_integration_points; ++ip)
+    {
+        pos.setIntegrationPoint(ip);
+
+        // \todo the argument to getValue() has to be changed for non
+        // constant storage model
+        auto const specific_storage =
+            _process_data.porous_media_properties.getSpecificStorage(t, pos)
+                .getValue(0.0);
+
+        auto const& ip_data = _ip_data[ip];
+        auto const& N = ip_data.N;
+        auto const& dNdx = ip_data.dNdx;
+        auto const& w = ip_data.integration_weight;
+
+        double C_int_pt = 0.0;
+        double p_int_pt = 0.0;
+        // Order matters: First C, then p!
+        NumLib::shapeFunctionInterpolate(local_x, N, C_int_pt, p_int_pt);
+
+        double const pc_int_pt = -p_int_pt;
+        double const Sw = (pc_int_pt > 0)
```

Comment:
You can remove this ternary condition because  (pc_int_pt > 0) is already checked inside 
```
_process_data.porous_media_properties
                           .getCapillaryPressureSaturationModel(t, pos)
                                 .getSaturation(pc_int_pt)
```
as
```
double BrooksCoreyCapillaryPressureSaturation::getSaturation(
    const double capillary_pressure) const
{
    const double pc =
        (capillary_pressure < 0.0) ? _minor_offset : capillary_pressure;
     ...
    return MathLib::limitValueInInterval(S, _saturation_r + _minor_offset,
                                         _saturation_max - _minor_offset);
```

Reply:
Thanks for the hint. Changed all locations.